test/e2e/fishapp/shared/manifest_builder.go

```diff
@@ -0,0 +1,275 @@
+package shared
+
+import (
+	"fmt"
+	appmeshv1beta1 "github.com/aws/aws-app-mesh-controller-for-k8s/pkg/apis/appmesh/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	defaultAppImage       = "970805265562.dkr.ecr.us-west-2.amazonaws.com/colorteller:latest"
+	defaultHTTPProxyImage = "abhinavsingh/proxy.py:latest"
+)
+
+const (
+	appContainerPort       = 9080
+	httpProxyContainerPort = 8899
+)
+
+type ServiceDiscoveryType int
+
+const (
+	DNSServiceDiscovery ServiceDiscoveryType = iota
```

Comment:
any reason not to use types in crd?

Reply:
i only find a meshServiceDiscoveryType defined, but there is only DNS(no cloudMap).